pypeit/core/tracewave.py

```diff
@@ -474,11 +498,12 @@ def trace_tilts(arcimg, lines_spec, lines_spat, thismask, slit_cen, inmask=None,
     # Do a PCA fit, which rejects some outliers
     iuse = trace_dict0['use_tilt']
     nuse = np.sum(iuse)
+    bpm = np.ones(trace_dict0['tilts_sub_fit'].shape, dtype=bool)
+    bpm[:,iuse] = False
     msgs.info('PCA modeling {:d} good tilts'.format(nuse))
-    pca_fit, poly_fit_dict, pca_mean, pca_vectors = extract.pca_trace(
-        trace_dict0['tilts_sub_fit'], predict=np.invert(iuse), npca=npca, coeff_npoly=coeff_npoly_pca,
-        lower=sigrej_pca, upper=sigrej_pca, order_vec=lines_spec, xinit_mean=lines_spec,
-        minv=0.0, maxv=float(trace_dict0['nsub'] - 1), debug=debug_pca)
```

Comment:
I think there may be a bug in your implementation of the PCA fitting. We need to always specify the minv maxv of the PCA fits. This is because robust_polyfit_djs requires not only the coefficients, but the min, max used for the fit. This is a bug in that implementation, i.e. robust_polyfit_djs should be returning an object with that information. At present, I'm worried that in your code when you use minx, maxx = None, you are basically performing a fit with one set of minx,maxx determined from the data and then evaluating that fit with a different set, determined from where you want to evaluate.

Reply:
Another great catch!  You're right that the min/max values weren't being propagated, particularly to the PCA predictions.  I've made a few edits in an attempt to fix this, and am testing.  The `TracePCA` object is an attempt to streamline this because the min/max parameters are kept as part of the class, and now explicitly used in its `predict` function.

Reply:
Yes, we need to also change robust_polyfit_djs to return an object that has this information. It is silly that these are input params.